src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -63,10 +65,25 @@ export const NavBarSchedule: FC<NavBarScheduleProps> = ({ isVisible }) => {
   const userEventIds =
     userWithId?.myPersonalizedSchedule ?? emptyPersonalizedSchedule;
 
-  const { isLoading, relatedVenues, relatedVenueIds } = useRelatedVenues({
+  const {
+    isLoading,
+    relatedVenues,
+    relatedVenueIds,
+    sovereignVenue,
+  } = useRelatedVenues({
     currentVenueId: venueId,
   });
 
+  const hasScheduleStartDate: boolean = isTruthy(
+    sovereignVenue?.scheduleStartDate
+  );
+  const scheduleStartDate: Date = sovereignVenue?.scheduleStartDate?.toDate()!;
+  const firstDay = useMemo(() => {
+    return hasScheduleStartDate
+      ? startOfDay(scheduleStartDate)
+      : startOfToday();
+  }, [hasScheduleStartDate, scheduleStartDate]);
```

Comment:
Why are you using `!` here? We don't use that feature of TypeScript as it basically breaks types (eg. the whole point of using TypeScript in the first place). In this instance, you're saying the value will always exist, which is explicitly untrue based on your use of `?.` before it. This will stop errors showing in TypeScript at coding time, but will just crash the platform at runtime.

For the sake of a few extra characters in the variable name, `firstDayOfSchedule` is more semantically explicit about its purpose than `firstDay`, which leads do a better developer experience (which is why I suggested that naming in my previous review)

This will correctly keep the type of `scheduleStartDate` as `Date | undefined`, and then use the 'falsiness`' of `undefined` in the ternary to choose which value to use.

```suggestion
  const scheduleStartDate = sovereignVenue?.scheduleStartData?.toDate();
 
  const firstDayOfSchedule = useMemo(() => {
    return scheduleStartDate
      ? startOfDay(scheduleStartDate)
      : startOfToday();
  }, [scheduleStartDate]);
```

Reply:
Thanks for pointing that out.

Reply:
No worries. I find that explaining the context/'why' of things allows people to make better proactive judgement calls on their own in future. Rather than me needing to provide the feedback each time.

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1419/commits/b78373170c83bbc19324c8694b04ea35b9ab06ef